clang/test/Driver/sycl-offload.c

```diff
@@ -712,6 +712,24 @@
 // RUN:   | FileCheck -check-prefix=CHK-TOOLS-IMPLIED-OPTS %s
 // CHK-TOOLS-IMPLIED-OPTS: clang-offload-wrapper{{.*}} "-compile-opts=-g -cl-opt-disable -DFOO1 -DFOO2"
 
+// RUN:   %clang -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice -g -O0 -Xsycl-target-backend "-DFOO1 -DFOO2" %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-TOOLS-IMPLIED-OPTS-FPGA %s
+// RUN:   %clang_cl -### -fsycl -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice -Zi -Od -Xsycl-target-backend "-DFOO1 -DFOO2" %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-TOOLS-IMPLIED-OPTS-FPGA %s
+// CHK-TOOLS-IMPLIED-OPTS-FPGA: aoc{{.*}} "-g" "-cl-opt-disable" "-DFOO1" "-DFOO2"
+
+// RUN:   %clang -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_x86_64-unknown-unknown-sycldevice -g -O0 -Xsycl-target-backend "-DFOO1 -DFOO2" %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-TOOLS-IMPLIED-OPTS-CPU %s
+// RUN:   %clang_cl -### -fsycl -fsycl-targets=spir64_x86_64-unknown-unknown-sycldevice -Zi -Od -Xsycl-target-backend "-DFOO1 -DFOO2" %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-TOOLS-IMPLIED-OPTS-CPU %s
+// CHK-TOOLS-IMPLIED-OPTS-CPU: opencl-aot{{.*}} "--bo=-g -cl-opt-disable" "-DFOO1" "-DFOO2"
```

Comment:
I wonder if "--bo=-g -cl-opt-disable" passed to CPU AOT really works, and -cl-opt-disable is not silently dropped? Maybe --bo=-g --bo=-cl-opt-disable ?
Just a question, if you checked and it works fine, then I'm OK :)

Reply:
I verified that it works:

> sycl-run:1258> clang++ -fsycl ~/a.cpp -fsycl-targets=spir64_x86_64-unknown-unknown-sycldevice  -O0 -g
Platform name: Intel(R) OpenCL
Device name: Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
Driver version: 2020.10.6.0.25.prerelease
OpenCL program was successfully created from SPIR-V file /tmp/a-00e640.spv
Using build options: -g -cl-opt-disable -I "/tmp"
Compilation started
Compilation done
Linking started
Linking done
Device build started
Options used by backend compiler: -g -cl-opt-disable
Device build done
Done.

Reply:
I don't know if this should be covered by our tests (probably not), but just as a double-check, does overriding these default options for offline compilers work as expected?

Reply:
I would assume that there is left to right option override behaviors.  If not, the issue would be with the respective tools.